### PR TITLE
Fixed KeepAlive Convergence test

### PR DESCRIPTION
**Pull request by bolt12**
# Description

This small PR just extracts a particular commit from https://github.com/input-output-hk/ouroboros-network/pull/3682 since now the actual PR lives in https://github.com/input-output-hk/io-sim/pull/9.

# Checklist

- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Commits have useful messages
    - [x] New tests are added if needed and existing tests are updated
    - [ ] If this branch changes Consensus and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../ouroboros-consensus/docs/interface-CHANGELOG.md)
    - [ ] If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)
    - [ ] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [x] Self-reviewed the diff
    - [x] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [x] Reviewer requested

### CODE REVIEW

File: ouroboros-network/test/Test/Ouroboros/Network/KeepAlive.hs [hunk 1]
```diff
@@ -155,27 +155,23 @@ prop_keepAlive_convergenceM tracer (NetworkDelay nd) seed = do
...
-    -- Can be removed after #2631 is merged.
-    threadDelay (keepAliveInterval + 128)
-
 -- Test that our estimate of PeerGSV's G terms converge to
 -- a given constant delay.
 prop_keepAlive_convergence :: NetworkDelay -> Int -> Property
 prop_keepAlive_convergence nd seed =
-    let trace = selectTraceEventsDynamic $ runSimTrace $ prop_keepAlive_convergenceM dynamicTracer nd seed in
-    verifyConvergence trace
+    let trace = take 1000
```
Comment (coot) [inline ouroboros-network/test/Test/Ouroboros/Network/KeepAlive.hs:170]: We used to wait `1000s` now we take first `1000` traces.  Any idea how they correspond to each other?

Comment (coot) [reply ouroboros-network/test/Test/Ouroboros/Network/KeepAlive.hs:170]: That's fine, I just wonder how much time moves in 1000 trace events.

Comment (bolt12) [reply ouroboros-network/test/Test/Ouroboros/Network/KeepAlive.hs:170]: I am not sure, I just put a big number to increase the amount of events we analyze so we can be more confident in the test. I think 1000s was rather arbitrary as well.

Comment (coot) [reply ouroboros-network/test/Test/Ouroboros/Network/KeepAlive.hs:170]: It's actually around `10000s`
```
  KeepAlive
    KeepAlive Convergence: OK (5.32s)
      +++ OK, passed 100 tests:
      15% 10600s
      12% 10400s
      12% 10700s
      10% 10200s
      10% 10500s
      10% 10800s
       9% 10000s
       8% 10900s
       7% 10100s
       6% 10300s
       1% 9900s
```

**Review by coot — APPROVED** [review]

**Comment by bolt12** [comment]
bors merge

File: ouroboros-network/test/Test/Ouroboros/Network/KeepAlive.hs (hunk 1 above)
Comment (bolt12) [reply ouroboros-network/test/Test/Ouroboros/Network/KeepAlive.hs:170]: Thank you!

**Pull request merged**